src/cart/services/cart.service.ts

```diff
@@ -1,15 +1,54 @@
 import { Injectable } from '@nestjs/common';
 
 import { v4 } from 'uuid';
+import { Client } from 'pg';
 
 import { Cart } from '../models';
 
+const dbProps = {
+  host: process.env.DB_HOST,
+  port: process.env.DB_PORT,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+  database: process.env.DB_NAME
+};
+
 @Injectable()
 export class CartService {
   private userCarts: Record<string, Cart> = {};
 
-  findByUserId(userId: string): Cart {
-    return this.userCarts[ userId ];
+  async findByUserId(userId: string): Promise<Cart> {
+    const client = new Client(dbProps);
+    await client.connect();
+
+    try{
+      const result = await client.query(`
+        SELECT *
+        FROM carts as c
+        LEFT JOIN cart_items AS ci on c.id = ci.cart_id
+        WHERE c.user_id = '${userId}';
+      `);
```

Comment:
it's better to have a separate data layer and hide all SQL in that layer, 
https://ctrly.blog/nodejs-layered-architecture/#:~:text=A%20small%20app%20consists%20of,a%20response%20from%20the%20server.